transformation/transformationPlugins/jsonMessageTransformers/jsonJinjavaTransformer/src/main/java/org/opensearch/migrations/transform/JinjavaTransformer.java

```diff
@@ -0,0 +1,90 @@
+package org.opensearch.migrations.transform;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+import org.opensearch.migrations.transform.jinjava.DynamicMacroFunction;
+import org.opensearch.migrations.transform.jinjava.JavaRegexCaptureFilter;
+import org.opensearch.migrations.transform.jinjava.JavaRegexReplaceFilter;
+import org.opensearch.migrations.transform.jinjava.NameMappingClasspathResourceLocator;
+import org.opensearch.migrations.transform.jinjava.ThrowTag;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hubspot.jinjava.Jinjava;
+import com.hubspot.jinjava.lib.fn.ELFunctionDefinition;
+import com.hubspot.jinjava.loader.ResourceLocator;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class JinjavaTransformer implements IJsonTransformer {
+
+    protected static final ObjectMapper objectMapper = new ObjectMapper();
+
+    protected final Jinjava jinjava;
+    protected final Function<Map<String, Object>, Map<String, Object>> createContextWithSourceFunction;
+    private final String templateStr;
+
+    public JinjavaTransformer(String templateString,
+                              UnaryOperator<Map<String, Object>> contextProviderFromSource) {
+        this(templateString, contextProviderFromSource, new NameMappingClasspathResourceLocator());
+    }
+
+    public JinjavaTransformer(String templateString,
+                              UnaryOperator<Map<String, Object>> createContextWithSource,
+                              ResourceLocator resourceLocator)
+    {
+        jinjava = new Jinjava();
+        this.createContextWithSourceFunction = createContextWithSource;
+        jinjava.setResourceLocator(resourceLocator);
+        jinjava.getGlobalContext().registerFilter(new JavaRegexCaptureFilter());
+        jinjava.getGlobalContext().registerFilter(new JavaRegexReplaceFilter());
+
+        jinjava.getGlobalContext().registerFunction(new ELFunctionDefinition(
+            "",
+            "invoke_macro",
+            DynamicMacroFunction.class,
+            "invokeMacro",
+            String.class,
+            Object[].class
+        ));
+        jinjava.getGlobalContext().registerTag(new ThrowTag());
```

Comment:
I think I got it. So these filters/functions/tags would become part of our formal interface specification to the LLM as options it can use when generating its Jinja, yeah?

Reply:
This one unifies the 'throw' tag between jinja2 and jinjava.  Without this tag, valid jinja that throws wouldn't be able to be interpreted with jinjava.  So, to answer your question, in some cases we might need to lift this configuration.  I hope that we don't have to though so that there's good parity between jinja and jinjava - so that we don't need to think about hosting a python runtime for these.

However, I DO think we'll definitely want to lift the jinja library files into the context so that those snippets can be leveraged and composited to create more consistent solutions.  Hopefully, there's less room for error too.